public/js/event.js

```diff
@@ -83,4 +312,73 @@ $(document).ready(function(){
     }
   });
 
+  var post = new Post();
+
+  $('#joinEventButton').click(function(){
```

Comment:
I'd recommend also to separate the API accessing functions and classes into a separate file, and then using `application.js` to manipulate the DOM calling those methods, rather than having everything in one big file